### ISSUE TITLE
Using username nad password for authentication leads to error

### ISSUE THREAD

**Issue by motabass** [nodeshift/openshift-rest-client#330]
I'm getting the following error when i try to use a config like this:

``` 
config = {
    url: process.env.OPENSHIFT_API_URL,
    auth: {
      username: process.env.OPENSHIFT_ADMIN_USERNAME,
      password: process.env.OPENSHIFT_ADMIN_PASSWORD
    },
    insecureSkipTlsVerify: true
  };
```

```/home/mohoritsch/WORKSPACE/customer.portal/mng-clients/node_modules/undici/lib/core/util.js:109
    throw new InvalidArgumentError('invalid url')
          ^
InvalidArgumentError: invalid url
    at Object.parseOrigin (/home/mohoritsch/WORKSPACE/customer.portal/mng-clients/node_modules/undici/lib/core/util.js:109:11)
    at new Client (/home/mohoritsch/WORKSPACE/customer.portal/mng-clients/node_modules/undici/lib/client.js:199:23)
    at /home/mohoritsch/WORKSPACE/customer.portal/mng-clients/node_modules/openshift-rest-client/lib/basic-auth-request.js:44:20
    at new Promise (<anonymous>)
    at getTokenFromBasicAuth (/home/mohoritsch/WORKSPACE/customer.portal/mng-clients/node_modules/openshift-rest-client/lib/basic-auth-request.js:40:10)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by lholmquist** [comment]
what version of the client are you using?

in the 7.1.x release we updated somethings, would could be the culprit.  if you are using 7.1.x, could you try the 7.0.0 release to see if that works.  If it does,  then that should narrow the problem down and we can get a fix out.   

Also,  if possible, could you share what the value of process.env.OPENSHIFT_API_URL is?

**Comment by mkalam-alami** [comment]
Hi,

I confirm the problem is related to 7.1.x, the stacktrace provided in the original post points to a problematic integration of unidici in [basic-auth-request.js](https://github.com/nodeshift/openshift-rest-client/blob/release-v7.2.0/lib/basic-auth-request.js#L44) (it seems like the lib expects a root URL when creating the client instead of a full URL with a pathname).

Version 7.0.0 works fine.

**Comment by lholmquist** [comment]
based on @mkalam-alami 's analysis,  i think i know how to fix this

**Comment by lholmquist** [comment]
hopefully the fix released in 8.0.1 will fix this
